src/server/scripts/Northrend/ChamberOfAspects/ObsidianSanctum/instance_obsidian_sanctum.cpp

```diff
@@ -40,6 +42,15 @@ class instance_obsidian_sanctum : public InstanceMapScript
             memset(&Encounters, 0, sizeof(Encounters));
         };
 
+        bool IsEncounterInProgress() const
+        {
+            for (uint8 i = 0; i < MAX_ENCOUNTER; ++i)
```

Comment:
//for (uint8 i = 0; i < MAX_ENCOUNTER; ++i)
for (uint8 i = 0; i < MAX_ENCOUNTERS; ++i)

Reply:
it's not blizzlike, because dragons haven't loot, and its not matter

Reply:
@ArashGorge 
What I don't understand here: "MAX_ENCOUNTER" is 1, so the loop will only be executed once for the value "0", why is the loop necessary at all? Wouldn't it suffice to use something like this instead of the loop?
```cpp
if (Encounters[BOSS_SARTHARION_EVENT] == IN_PROGRESS)
    return true;
```

Reply:
I tested it, didn't work
if i remember

Reply:
Hmm, that's strange, because semantically it is the same. Could you please tell me how to test this?

Reply:
as i say i don't remember , but it should work (tested and works fine), and your codes maybe i wrong in testing event , I don't think there is a problem with my write's ,but if you want i can test again and i'll told result as both of them